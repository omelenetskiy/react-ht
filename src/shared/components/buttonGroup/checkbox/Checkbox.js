import * as React from 'react';

import StyledCheckbox from '../../../styled/checkbox';

type Props = {
  button: {
    value: string,
    title: string,
    name: string,
    checked: boolean,
  },
  toggleSearchBy: SyntheticEvent<>,
};

const Checkbox = ({ button, toggleSearchBy }: Props) => (
  <StyledCheckbox
    type="radio"
    value={button.value}
    id={button.title}
    name={button.name}
    defaultChecked={button.checked}
    onChange={toggleSearchBy}
  />
);

export default Checkbox;
