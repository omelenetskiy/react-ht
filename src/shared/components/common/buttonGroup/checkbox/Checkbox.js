import React from 'react';

import StyledCheckbox from '../../../styled/checkbox';

const Checkbox = ({ button, toggleSearchBy }) => (
  <StyledCheckbox
    type="radio"
    value={button.value}
    id={button.title}
    name={button.name}
    defaultChecked={button.checked}
    onChange={toggleSearchBy}
  />
);

export default Checkbox;
