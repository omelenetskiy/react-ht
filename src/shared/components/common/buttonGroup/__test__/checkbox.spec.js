import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import Checkbox from '../checkbox/Checkbox';

const props = {
  button: {
    value: 'value',
    title: 'title',
    name: 'name',
    checked: true
  },
  toggleSearchBy: jest.fn()
};

describe('Checkbox', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Checkbox {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
