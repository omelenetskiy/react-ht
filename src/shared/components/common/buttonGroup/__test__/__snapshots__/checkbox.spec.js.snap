// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<checkbox___default
  defaultChecked={true}
  id="title"
  name="name"
  onChange={[MockFunction]}
  type="radio"
  value="value"
/>
`;
